server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1364,5 +1364,12 @@ public void calculateExportedForOwnersPools(Owner owner) {
         q.executeUpdate();
     }
 
-
+    public void markCertificatesDirtyForOwnerProductId(Owner owner, String productId) {
```

Comment:
Pedantic request: can we rename this to ```markCertificatesDirtyFor(Pools|Entitlements)WithProduct``` (Select whichever is preferred for the noun before "with")?